buildconfigs/06-cli.yaml

```diff
@@ -9,7 +9,7 @@ spec:
     type: Git
     git:
       uri: 'https://github.com/openshift/oc'
-      ref: release-4.12
+      ref: master
```

Comment:
I'd rather use release-4.15 here. Master may move further and its useful to point to a BC at some point in time

Reply:
Hey @vrutkovs , I was thinking of having a `release-4.15` for it instead. But I'm not sure, let's have a decision about this... I could even move some changes to another PR, although I prefer avoiding it if we could (too many changes here to split).

Reply:
I'm okay with masters if we'll have versions for `release-4.x`. But at this point `master` and `release-4.15` are identical, while later these would require more modification, effectively making it "master as of end of 2023". That might bite us later

Reply:
After this whole repo becomes a stable thing, I would like we maintain a 'stable' branch for the current release and master for the 'next' one.

Do you prefer doing as in branches not having the name of the release (stable and master)?

Reply:
Sounds good. I think focusing on two (stable and next) is reasonable

Reply:
do you prefer a `master` branch tracking the `master` branch of components and a `stable` one tracking `release-4.x`, with x being the current release? or, `master` branch tracking `release-4.${next}` and `stable` tracking `release-4.${current}`?

Reply:
I think `next` tracking `release-4.15` and `stable` tracking `release-4.14` would be the best option - this way we be sure that at any given time the list of images doesn't change

Reply:
Fixing in #60

Reply:
I'll drop half of the commits related to the 4.15 'branching' in this PR, to focus on the Argo workflows changes.

Reply:
done